### ISSUE TITLE
Invalid check argument errors

### ISSUE THREAD

**Issue by cgmartin** [sensu-plugins/sensu-plugins-fluentd#1]
Seeing `Invalid check arguments(s)` errors with what seems like correct arguments...

```
# /usr/local/bin/check-fluentd-monitor-agent.rb -w 10 -c 20 --metric retry_count
Invalid check argument(s): invalid option: -w, ["/usr/local/share/gems/gems/mixlib-cli-1.5.0/lib/mixlib/cli.rb:191:in `parse_options'", "/usr/local/share/gems/gems/sensu-plugin-1.2.0/lib/sensu-plugin/cli.rb:13:in `initialize'", "/usr/local/share/gems/gems/sensu-plugin-1.2.0/lib/sensu-plugin/cli.rb:55:in `new'", "/usr/local/share/gems/gems/sensu-plugin-1.2.0/lib/sensu-plugin/cli.rb:55:in `block in <class:CLI>'"]
/usr/share/ruby/net/protocol.rb:52:in `<module:Net>': uninitialized constant Timeout::Error (NameError)
    from /usr/share/ruby/net/protocol.rb:24:in `<top (required)>'
    from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:126:in `require'
    from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:126:in `require'
    from /usr/share/ruby/net/http.rb:22:in `<top (required)>'
    from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:126:in `require'
    from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:126:in `require'
    from /usr/local/share/gems/gems/sensu-plugins-fluentd-0.0.1/bin/check-fluentd-monitor-agent.rb:30:in `<top (required)>'
    from /usr/local/bin/check-fluentd-monitor-agent.rb:23:in `load'
    from /usr/local/bin/check-fluentd-monitor-agent.rb:23:in `<main>'
```

Using gem: sensu-plugins-fluentd (0.0.1)

**Comment by stanhu** [comment]
I'm also seeing his if I install via the gem. If I run `/opt/sensu/embedded/bin/check-fluentd-monitor-agent.rb`, I see:

```
Sensu::Plugin::CLI: ["Not implemented! You should override Sensu::Plugin::CLI#run."]
/opt/sensu/embedded/lib/ruby/2.0.0/net/protocol.rb:52:in `<module:Net>': uninitialized constant Timeout::Error (NameError)
        from /opt/sensu/embedded/lib/ruby/2.0.0/net/protocol.rb:24:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/net/http.rb:22:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugins-fluentd-0.0.1/bin/check-fluentd-monitor-agent.rb:30:in `<top (required)>'
        from ./check-fluentd-monitor-agent.rb:23:in `load'
        from ./check-fluentd-monitor-agent.rb:23:in `<main>'
```

But if I run  /`/opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugins-fluentd-0.0.1/bin/check-fluentd-monitor-agent.rb` directly, everything works. 

The gem-installed `bin/check-fluentd-monitor-agent.rb` looks like:

```
#!/opt/sensu/embedded/bin/ruby
#
# This file was generated by RubyGems.
#
# The application 'sensu-plugins-fluentd' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'rubygems'

version = ">= 0"

if ARGV.first
  str = ARGV.first
  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
  if str =~ /\A_(.*)_\z/
    version = $1
    ARGV.shift
  end
end

gem 'sensu-plugins-fluentd', version
load Gem.bin_path('sensu-plugins-fluentd', 'check-fluentd-monitor-agent.rb', version)
```

**Comment by eheydrick** [comment]
Can you test with the fix that was just merged?

**Comment by stanhu** [comment]
That didn't seem to solve the problem for me. It looks like the issue is that `net/protocol.rb` is not finding `Timeout::Error`.

**Comment by eheydrick** [comment]
Can you post the full command line and output. Also can you test the latest master version - I just pushed a fix for the timeout variable being a string vs an integer. It's successfully running for me.

**Comment by stanhu** [comment]
Same as I posted earlier:

```
$ /opt/sensu/embedded/bin/check-fluentd-monitor-agent.rb 
Sensu::Plugin::CLI: ["Not implemented! You should override Sensu::Plugin::CLI#run."]
/opt/sensu/embedded/lib/ruby/2.0.0/net/protocol.rb:52:in `<module:Net>': uninitialized constant Timeout::Error (NameError)
        from /opt/sensu/embedded/lib/ruby/2.0.0/net/protocol.rb:24:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/net/http.rb:22:in `<top (required)>'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugins-fluentd-0.0.1/bin/check-fluentd-monitor-agent.rb:30:in `<top (required)>'
        from /opt/sensu/embedded/bin/check-fluentd-monitor-agent.rb:23:in `load'
        from /opt/sensu/embedded/bin/check-fluentd-monitor-agent.rb:23:in `<main>'
```

Running the script by hand (i.e. `/opt/sensu/embedded/bin/ruby  /opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/sensu-plugins-fluentd-0.0.1/bin/check-fluentd-monitor-agent.rb```) works still.

I suspect there's a conflict with the timeout gem in `/opt/sensu/embedded/lib/ruby/gems/2.0.0/gems/timeout-0.0.1/lib/timeout.rb`.

**Comment by eheydrick** [comment]
You are right, the problem is a conflict between the timeout gem and the built-in timeout. I'm thinking we should drop the gem dependency and just use the built-in one. Same issue as https://github.com/sensu-plugins/sensu-plugins-ponymailer/issues/3.

**Comment by eheydrick** [comment]
There was a problem with the deployment and new versions weren't getting pushed to rubygems. I fixed that and pushed version [0.1.0](https://rubygems.org/gems/sensu-plugins-fluentd/versions/0.1.0), which contains fixes for the timeout issue.
